mongo/mongo.go

```diff
@@ -27,3 +35,42 @@ func (m *Mongo) Init() (session *mgo.Session, err error) {
 	session.SetMode(mgo.Strong, true)
 	return session, nil
 }
+
+// GetRecipes retrieves all recipe documents from Mongo
+func (m *Mongo) GetRecipes() ([]recipe.Response, error) {
+	s := m.Session.Copy()
+	defer s.Close()
+
+	iter := s.DB(m.Database).C(m.Collection).Find(nil).Iter()
+	defer func() {
+		err := iter.Close()
+		if err != nil {
+			log.Event(context.Background(), "error closing iterator", log.ERROR, log.Error(err))
+		}
+	}()
+
+	results := []recipe.Response{}
+	if err := iter.All(&results); err != nil {
+		if err == mgo.ErrNotFound {
+			return nil, errs.ErrRecipeNotFound
```

Comment:
I have added a new error "ErrRecipesNotFound" in `apierrors` package

When dealing with empty list, as  `items` and `total_count` are set as `omitempty`, then total_count would be set to 0 but it would not be printed because of the omitempty. This is the same with the previous functionality when mongo was not in place. 

I can change the structure so that the `total_count` is a pointer in which the value will be pointed and thus printed. But this is changing the existing functionality and need to change the data.go file. We want it so that we don't want to make any changes to the existing functionality and working the same with mongo and it is doing so.

Reply:
Hmmm, well argued but the existing functionality is wrong, total_count can be 0 and should return 0 in json document; so you are right in that it needs to be a pointer to achieve this.

This can be done another time to allow empty items array - this is the same issue with the import api; so something to look at another time.